vars/buildJenkinsComponent.groovy

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env groovy
```

Comment:
cannot comment on the filename, and need to review this more fully some day, but I think this should be named  `buildComponent`.
This would be more consistent with `buildPlugin`, and like for the plugins where we removed Jenkins because it's contextually obviously Jenkins related, I think this is the same here :).